src/main/scala/diplomacy/LazyModule.scala

```diff
@@ -117,11 +122,14 @@ abstract class LazyModule()(implicit val p: Parameters)
     children.foreach( _.nodeIterator(iterfunc) )
   }
 
-  def getChildren = children
+  def getChildren() = children
+//  def getNodes() = nodes
+
 }
 
 object LazyModule
 {
+  val impToLazyModuleMap = new mutable.HashMap[String, (LazyModuleImp, LazyModule)]
```

Comment:
All `LazyModuleImpLike`'s have a `wrapper` field. Why can't we reconstruct this `impToLazyModuleMap` from the pointer to the top-level module's wrapper?

Reply:
can you add a comment for this? What is the `String` used to lookup here?

@azidar are you saying `wrapper` already gives a mapping from any LazyModuleImpLink back to its instantiating `LazyModule`?